tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -106,488 +95,446 @@ describe(__filename, () => {
+    const user = getCurrentUser(store.getState().users);
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'addonType',
-      ADDON_TYPE_STATIC_THEME,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pageParam',
-      'page_t',
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp('paginate', true);
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pathname',
-      `/user/${userId}/`,
-    );
+    expect(screen.getByText(`Themes by ${user.name}`)).toBeInTheDocument();
+    const themeCard = screen.queryAllByClassName('AddonsByAuthorsCard').item(1);
+    expect(screen.queryByText(themeCard, 'Previous')).not.toBeInTheDocument();
+    expect(screen.queryByText(themeCard, 'Next')).not.toBeInTheDocument();
   });
 
   it('renders a not found page if the API request is a 404', () => {
-    const { store } = dispatchSignInActions();
-    const errorHandler = new ErrorHandler({
-      id: 'some-error-handler-id',
-      dispatch: store.dispatch,
-    });
-    errorHandler.handle(
-      createApiError({
+    const userId = 123;
+    createFailedErrorHandler({
+      error: createApiError({
         response: { status: 404 },
         apiURL: 'https://some/api/endpoint',
         jsonResponse: { message: 'not found' },
       }),
-    );
+      id: createErrorHandlerId({ userId }),
+      store,
+    });
 
-    const root = renderUserProfile({ errorHandler, store });
+    renderUserProfile({ params: { userId } });
 
-    expect(root.find(NotFoundPage)).toHaveLength(1);
+    expect(
+      screen.getByText('Oops! We can’t find that page'),
+    ).toBeInTheDocument();
   });
 
   it('renders errors', () => {
-    const { store } = dispatchSignInActions();
-    const errorHandler = new ErrorHandler({
-      id: 'some-id',
-      dispatch: store.dispatch,
+    const userId = 123;
+    const errorString = 'unexpected error';
+    createFailedErrorHandler({
+      error: new Error(),
+      id: createErrorHandlerId({ userId }),
+      message: errorString,
+      store,
     });
-    errorHandler.handle(new Error('unexpected error'));
 
-    const root = renderUserProfile({ errorHandler, store });
+    renderUserProfile({ params: { userId } });
 
-    expect(root.find(ErrorList)).toHaveLength(1);
+    expect(screen.getAllByText(errorString)).not.toHaveLength(0);
   });
 
   it('renders an edit link', () => {
-    const root = renderUserProfile();
+    signInUserWithProps({ userId: 100 });
+    renderUserProfile();
 
-    expect(root.find('.UserProfile-edit-link')).toHaveLength(1);
-    expect(root.find('.UserProfile-edit-link')).toHaveProp('to', `/users/edit`);
-    expect(root.find('.UserProfile-edit-link').children()).toHaveText(
-      'Edit profile',
+    const editButton = screen.getByRole('link', { name: 'Edit profile' });
+    expect(editButton).toBeInTheDocument();
```

Comment:
This was mentioned above, and applies to all tests in this file: We don't need to assert that something is in the document if we are also finding it via a `getBy`, because the `getBy` will fail. Just combine it all into one statement that does the `getBy` and calls `toHaveAttribute`.